src/libs/antares/study/parts/hydro/series.h

```diff
@@ -144,108 +151,34 @@ class DataSeriesHydro
     */
     Matrix<double, int32_t> maxHourlyPumpPower;
 
-    /*!
-    ** \brief The number of time-series
-    **
-    ** This value must be the same as the width of the matrices `mod` and `fatal`.
-    ** It is only provided for convenience to avoid same strange and ambiguous code
-    ** (for example using `fatal.width` and `mod.width` in the same routine, it might
-    ** indicate that the two values are not strictly equal)
-    */
-    uint count;
+    unsigned int TScount() const { return generationTScount_; };
 
 public:
-    /*!
-    ** \brief Monte-Carlo
-    */
     Matrix<uint32_t> timeseriesNumbers;
     Matrix<uint32_t> timeseriesNumbersHydroMaxPower;
 
-    /**
-     * \brief Loading hydro maximum generation and pumping TS's
-     */
-    bool LoadMaxPower(const AreaName& areaID, const AnyString& folder);
-
-    /**
-     * \brief Post processing numbers of TS's
-     *  Checking whether or not TS's numbers are different
-     *  and taking proper action based on corresponding check
-     */
-    bool postProcessMaxPowerTS(Area& area);
-
-    void setHydroModulability(Study& study, const AreaName& areaID) const;
-
-    /**
-     * \brief Setting TS's number of maximum generation and pumping matrices
-     */
-    void setNbTimeSeriesSup(uint nbTimeSeriesSup);
+    // Equalizing max generation and max pumping numbers of TS's    
+    void EqualizeMaxPowerTSsizes(Area& area, bool& fatalError);
 
-    /**
-     * \brief Setting TS's number of maximum generation and pumping matrices
-     */
-    void setNbTimeSeriesSup();
+    void setHydroModulability(Area& area) const;
 
-    /**
-     * \brief Get function for TS's number of maximum generation and pumping matrices
-     */
-    uint getNbTimeSeriesSup() const;
+    // Get max power (generatoin and pumping) number of TS
+    uint maxPowerTScount() const;
 
-    /**
-     * \brief Setting TS's when derated mode is on
-     */
-    void setMaxPowerTSWhenDeratedMode(const Study& study);
+    // Setting TS's when derated mode is on
+    void resizeTSinDeratedMode(bool derated, unsigned int studyVersion);
 
 private:
-    /*!
-    ** \brief The number of time-series for maxHourlyGenPower and maxHourlyPumpPower matrices
-    **
-    ** This value must be the same as the width of the matrices `maxHourlyGenPower` and
-    *`maxHourlyPumpPower`.
-    ** It is only provided for convenience to avoid same strange and ambiguous code
-    ** (for example using `maxHourlyGenPower.width` and `maxHourlyPumpPower.width` in the same
-    *routine, it might
-    ** indicate that the two values are not strictly equal)
-    */
-    uint nbTimeSeriesSup_ = 0;
-
-    /**
-     * This class provides comparing two integers
-     * that represents width of two corresponding matrices.
-     */
-    class NbTsComparer
-    {
-    public:
-        NbTsComparer(uint32_t nbOfGenPowerTs, uint32_t nbOfPumpPowerTs);
-
-        bool bothZeros() const;
```

Comment:
I removed the **bothZero** case because : 
Before calling **EqualizeTSsize** function, we always call a load function (**loadROR**, **loadMINGEN**, ...).
Each of them contain a call to **Matrix::loadFromCSVFile(...)**. And this function always gives to the matrix a width at least equal to one. So, in **EqualizeTSsize(...)**, even the case where one of the 2 matrices has a width = 0 is impossible.
Did I miss something ? Did you have a crash or a bug with GUI while loading hydro TS ?

Reply:
You are absolutely right what you have said, but that is the case when a study is loaded via solver. In case when we are opening (loading) a study via GUI, matrices will not be loaded and we are performing checks for empty matrices. How I know that they are empty when opening study via GUI? I've debug it. This is a snapshot of debug mode, where you can see that ror and storage matrices are empty after loading via GUI. 

![image](https://github.com/AntaresSimulatorTeam/Antares_Simulator/assets/131546833/d36e9ba1-2a69-4793-ad65-d44685d01e59)

I know that we are deprecating this GUI, but should we leave it this way?

Reply:
No, you're right, we don't want any bug / crash in GUI.
Thanks for having tested this.
I'm currently correcting this.

Reply:
Just to add, this will not crash GUI, but nevertheless it should be considered.